drop-storage/src/lib.rs

```diff
@@ -707,24 +710,27 @@ impl Storage {
 
         let mut conn = self.conn.acquire().await?;
 
-        let mut paths = sqlx::query!("SELECT * FROM incoming_paths WHERE transfer_id = ?1", tid)
-            .fetch_all(&mut *conn)
-            .await?
-            .into_iter()
-            .map(|p| IncomingPath {
-                id: p.id,
-                transfer_id,
-                relative_path: p.relative_path,
-                file_id: p.path_hash,
-                bytes: p.bytes,
-                created_at: p.created_at.timestamp_millis(),
-                pending_states: vec![],
-                started_states: vec![],
-                cancel_states: vec![],
-                failed_states: vec![],
-                completed_states: vec![],
-            })
-            .collect::<Vec<_>>();
+        let mut paths = sqlx::query!(
+            r#"SELECT id as "path_id!", * FROM incoming_paths WHERE transfer_id = ?1"#,
```

Comment:
Can you clarify the syntax a bit? It looks unfamiliar

Reply:
It's the sqlx thing. It treated the id as `Option<>` which is not possible because it's `NOT NULL`. So I used the sqlx syntax to forcibly make it not an `Option<>`. It's a kind of `unwrap()` for sqlx

Reply:
How is `id` an Option when it's directly assigned to `IncomingPath.id` ? I don't really understand this id rename

Reply:
Exactly that's the problem that this is an `Option`. It's just sqlx decided for some reason that it's an Option, not me.
I mean I could've used `id as "id!"` here as well but the point is I needed to use the `!` to force sqlx into not making `Option` out of it

Reply:
It happens after I added the `UNIQE` constraint. Sqlx somehow cannot prove to itself that id cannot be NULL after the check is added